enumerables.rb

```diff
@@ -0,0 +1,166 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |x|
+      yield(self[x])
+    end
+    self
+  end
+
+  def my_each_with_index(value = 0)
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = value
+    size.times do |x|
+      yield(self[x], index)
+      index += 1
+    end
+  end
+
+  def my_select(arg = nil)
+    return to_enum(:my_select) unless block_given? || !arg.nil?
+
+    arr = []
+
+    if !arg.nil?
+      my_each do |x|
+        arr.push(x) if arg(x)
+      end
+    else
+      my_each do |x|
+        arr.push(x) if yield(x)
+      end
+    end
+    arr
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_all?(arg = nil)
+    return false unless block_given? || arg.nil? == false || empty?
+
+    if block_given?
+      my_each do |x|
+        return false unless yield(x)
+      end
+    elsif arg.class == Regexp
+      my_each do |x|
+        return false unless x.match(arg)
+      end
+    elsif arg.class == Numeric
+      my_each do |x|
+        return false unless x.class == arg
+      end
+    else
+      my_each do |x|
+        return false unless arg == x
+      end
+    end
+    true
+  end
```

Comment:
# my_all?
1. [ ] Returns false if the block returns false or nil:: range
### Test 👉 `range.my_all?(&false_block)`
Resulted in *NoMethodError* 
#
2. [ ] When no block or argument is given returns true when none of the collection members is false or nil
### Test 👉 `true_array.my_all?`

|Expected|Got|
|:---:|:---:|
|`true`|`false`|
#
3. [ ] When a class is passed as an argument returns true if all of the collection is a member of such class:: Integer
### Test 👉 `array.my_all?(Integer)`

|Expected|Got|
|:---:|:---:|
|`true`|`false`| 
#
4. [ ] When a class is passed as an argument returns true if all of the collection is a member of such class:: Numeric
### Test 👉 `numbers.my_all?(Numeric)`

|Expected|Got|
|:---:|:---:|
|`true`|`false`|